### PR TITLE
#2142. Add syntax tests

### PR DESCRIPTION
**Pull request by sgrekhov**
It turns out that it's easier to add new etxtension types tests (which are, in fact, edited copies of inline classes tests) rather then rename and change inline classes ones. I'm going to go on this way and when there will be nothing more to copy and edit I'll delete Inline classes folder

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good! Just one thing I believe must be changed: `ET.new(42)` is not an error when `ET` declares a constructor (a primary constructor in this case, but any constructor would do) whose name is `ET`.

I also mentioned a couple of places where a test will need to be adjusted in the future if we adopt the primary constructor proposal (in its current form or something that isn't too different). I don't know what's the best approach; perhaps we can just ignore this situation and rely on our ability to rediscover it when needed? .. another approach could be to put a `TODO` comment saying that "when we have primary constructors, this needs to change as follows: ...". WDYT?

File: LanguageFeatures/Extension-types/syntax_A04_t04.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/// `representationDeclaration` in the form `identifier`.new is instantiated as
+/// `identifier`.new
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type ET.new(int id) {}
+
+main() {
+  ET et = ET.new(42);
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Extension-types/syntax_A04_t04.dart]: Indeed. I was sure that `<Class name>.new` syntax is used for the reference to a tear-off of the default constructor but not for invocation. But it isn't so. I'd should check it by myself first. Thank you for pointing. Added an apropriate call to a previous test and deleted this one

File: LanguageFeatures/Extension-types/syntax_A04_t06.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+///
+/// <representationDeclaration> ::=
+///   ('.' <identifierOrNew>)? '(' <metadata> <type> <identifier> ')'
+///
+/// <identifierOrNew> ::= <identifier> | 'new'
+///
+/// <extensionTypeMemberDeclaration> ::= <classMemberDefinition>
+///
+/// @description Checks that it is a compile-time error if `type` in
+/// `representationDeclaration` is `var` or `final`
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Extension-types/syntax_A04_t06.dart:23]: Added `TODO`

File: LanguageFeatures/Extension-types/syntax_A07_t01.dart [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+///
+/// <extensionTypeMemberDeclaration> ::= <classMemberDefinition>
+///
+/// @description Checks that it is a compile-time error if an extension type
+/// declaration has no body
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type ET(int id);
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Extension-types/syntax_A07_t01.dart:29]: Added `// TODO` again

**Review by sgrekhov — COMMENTED** [review]
Updated. Added `TODO` for the tests that may be affected by the primary constructors

File: LanguageFeatures/Extension-types/syntax_A04_t04.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Extension-types/syntax_A04_t04.dart]: Thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**